src/Compilers/Core/Portable/MetadataReader/MetadataDecoder.cs

```diff
@@ -1844,60 +1869,17 @@ protected TypeSymbol DecodeFieldSignature(ref BlobReader signatureReader, out bo
 
             try
             {
-                // See if there is a Volatile modifier.
+                var isVolatileFound = false;
                 SignatureTypeCode typeCode;
-                ArrayBuilder<ModifierInfo<TypeSymbol>> customModifierBuilder = null;
 
-                for (;;)
+                customModifiers = DecodeModifiersOrThrow(ref signatureReader, out typeCode, acceptRequiredModifier: type =>
```

Comment:
>DecodeModifiersOrThrow [](start = 34, length = 22)

This was refactored since it shares the same behavior as DecodeModifiersOrThrow(), making use of the additional parameter.